Sources/AIProxy/OpenAI/OpenAITextToSpeechRequestBody.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  
+//  OpenAITextToSpeechCodables.swift
+//
+//
+//  Created by Daniel Aditya Istyana on 10/9/24.
+//
+
+
+import Foundation
+
+public struct OpenAITextToSpeechRequestBody: Encodable {
+
+    // Required
+    /// The text to generate audio for. The maximum length is 4096 characters.
+    public let input: String
+
+    // Required
+    /// One of the available TTS models: `tts-1` or `tts-1-hd`, default to `tts-1`
+    public let model: Model
+
+    // Required
+    /// The voice to use when generating the audio. Supported voices are `alloy`, `echo`, `fable`, `onyx`, `nova`, and `shimmer`.
+    public let voice: Voice
+
+    /// The format to audio in. Supported formats are `mp3`, `opus`, `aac`, `flac`, `wav`, and `pcm`.
+    public let responseFormat: ResponseFormat
```

Comment:
`responseFormat` will serialize to `responseFormat` (which OpenAI does not understand) instead of `response_format`. So you have two options here:

1. Use JSONEncoder with `KeyEncodingStrategy.convertToSnakeCase` set. You would use this on line 206 of OpenAIService.swift.

2. Add coding keys for this struct.

Option 2 is my preferred approach, even though it's more work. I like it because it encapsulates the knowledge of how this type should be serialized in a single place. You can imagine someone using this type without a serializer that we've written (one that uses snake case) and accidentally getting invalid JSON. So, in this case, I would do this:

```
public struct OpenAITextToSpeechRequestBody: Encodable {
    // Your existing properties as-is

    // Add this:
    private enum CodingKeys: String, CodingKey {
        case input
        case model
        case responseFormat = "response_format"
        case speed
        case voice
    }
}
```

Reply:
Ah, my bad. This should be fixed in [0026817](https://github.com/lzell/AIProxySwift/pull/55/commits/0026817be9917a9bb24f6a588af881c9d13b7f78)